lib/flux-core/src/iter/traits/iterator.rs

```diff
@@ -0,0 +1,33 @@
+use flux_attrs::*;
+
+#[extern_spec(core::iter)]
+#[assoc(fn valid_item(self: Self, item: Self::Item) -> bool { true })]
+#[assoc(fn size(self: Self) -> int)]
+#[assoc(fn done(self: Self) -> bool)]
+#[assoc(fn step(self: Self, other: Self) -> bool { true })]
+trait Iterator {
+    #[flux::sig(fn(self: &strg Self[@curr_s]) -> Option<Self::Item>[!<Self as Iterator>::done(curr_s)] ensures self: Self{next_s: <Self as Iterator>::step(curr_s, next_s)})]
+    fn next(&mut self) -> Option<Self::Item>;
+
+    #[flux::sig(fn(Self[@s]) -> Enumerate<Self>[0, s])]
+    fn enumerate(self) -> Enumerate<Self>
+    where
+        Self: Sized;
+
+    #[flux::sig(fn(Self[@s], f: F) -> Map<Self, F>[s] where F: FnMut(Self::Item{item: <Self as Iterator>::valid_item(s, item)}) -> B)]
```

Comment:
Same here and other places

```suggestion
    #[flux::sig(
        fn(Self[@s], f: F) -> Map<Self, F>[s]
        where 
            F: FnMut(Self::Item{item: <Self as Iterator>::valid_item(s, item)}) -> B
    )]
```